core/src/main/java/com/linecorp/armeria/internal/metric/RequestMetricSupport.java

```diff
@@ -33,63 +33,50 @@
 import io.micrometer.core.instrument.DistributionSummary;
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Timer;
-import io.netty.util.AttributeKey;
 
 /**
  * Collects the metric data and stores it into the {@link MeterRegistry}.
  */
 public final class RequestMetricSupport {
 
-    private static final AttributeKey<RequestMetrics> ATTR_REQUEST_METRICS =
-            AttributeKey.valueOf(RequestMetricSupport.class, "REQUEST_METRICS");
-
-    private static final RequestMetrics PLACEHOLDER = new RequestMetricsPlaceholder();
-
     public static void setup(RequestContext ctx, MeterIdPrefixFunction meterIdPrefixFunction) {
-        if (ctx.hasAttr(ATTR_REQUEST_METRICS)) {
-            return;
-        }
-
-        // Set the attribute to the placeholder so that calling this method again has no effect.
-        // The attribute will be set to the real one later in onRequestStart()
-        ctx.attr(ATTR_REQUEST_METRICS).set(PLACEHOLDER);
-
         ctx.log().addListener(log -> onRequestStart(log, meterIdPrefixFunction),
                               RequestLogAvailability.REQUEST_HEADERS,
                               RequestLogAvailability.REQUEST_CONTENT);
-        ctx.log().addListener(RequestMetricSupport::onRequestEnd,
-                              RequestLogAvailability.REQUEST_END);
-        ctx.log().addListener(RequestMetricSupport::onResponse,
-                              RequestLogAvailability.COMPLETE);
     }
 
     private static void onRequestStart(RequestLog log, MeterIdPrefixFunction meterIdPrefixFunction) {
         final RequestContext ctx = log.context();
         final MeterRegistry registry = ctx.meterRegistry();
         final MeterIdPrefix idPrefix = meterIdPrefixFunction.apply(registry, log);
-        final RequestMetrics requestMetrics =  MicrometerUtil.register(
-                registry, idPrefix, RequestMetrics.class, DefaultRequestMetrics::new);
-
-        ctx.attr(ATTR_REQUEST_METRICS).set(requestMetrics);
-        requestMetrics.active().incrementAndGet();
+        final MeterIdPrefix idPrefixActive = new MeterIdPrefix(idPrefix.name("activeRequests"),
+                                                               idPrefix.tags());
+
+        ActiveRequestMetrics activeRequestMetrics = MicrometerUtil.register(
+                registry, idPrefixActive, ActiveRequestMetrics.class,
+                (reg, prefix) ->
+                        reg.gauge(prefix.name(), prefix.tags(),
+                                  new ActiveRequestMetrics(), ActiveRequestMetrics::doubleValue));
```

Comment:
We could then just replace `ActiveRequestMetrics` with `LongAdder` completely? e.g. `new LongAdder(), LongAdder::doubleValue`?

Reply:
It seems that MicrometerUtil will use class type as identification for mapping with MeterIdPrefix, so I thought it's better to not use std type like LongAdder, or it will prevent us to register further LongAdder. HDYT?

Reply:
SGTM